java/ql/src/experimental/Security/CWE/CWE-348/UseOfLessTrustedSource.ql

```diff
@@ -0,0 +1,52 @@
+/**
+ * @name IP address spoofing
+ * @description The software obtains the client ip from the remote endpoint identifier specified (`X-Forwarded-For`,
+ *              `X-Real-IP`, `Proxy-Client-IP`, etc.) in the header and uses it. Attackers can modify these The value
+ *              of the identifier to forge the client ip.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id java/ip-address-spoofing
+ * @tags security
+ *       external/cwe/cwe-348
+ */
+
+import java
+import UseOfLessTrustedSourceLib
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+/**
+ * Taint-tracking configuration tracing flow from obtain client ip to use the client ip.
+ */
+class UseOfLessTrustedSourceConfig extends TaintTracking::Configuration {
+  UseOfLessTrustedSourceConfig() { this = "UseOfLessTrustedSourceConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof UseOfLessTrustedSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof UseOfLessTrustedSink }
+
+  /**
+   * When using `,` split request data and not taking the first value of the array, it is considered as `good`.
```

Comment:
```suggestion
   * Splitting a header value by `,` and taking an entry other than the first is sanitizing, because
   * later entries may originate from more-trustworthy intermediate proxies, not the original client.
```